protocol/x/feetiers/keeper/msg_server.go

```diff
@@ -26,7 +26,7 @@ func (k msgServer) UpdatePerpetualFeeParams(
 	goCtx context.Context,
 	msg *types.MsgUpdatePerpetualFeeParams,
 ) (*types.MsgUpdatePerpetualFeeParamsResponse, error) {
-	if !k.HasAuthority(msg.Authority) {
+	if !k.Keeper.HasAuthority(msg.Authority) {
```

Comment:
nit: Is this change needed given the keeper type is embedded into the `msgServer` struct?

Reply:
I had to change `FeeTiersKeeper` in `App` to a pointer (and thus keeper is an attribute of `msgServer` rather than being embedded) and that necessitated this change